common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -498,6 +504,56 @@ public Object run() {
                 DIRECT_BUFFER_CONSTRUCTOR != null ? "available" : "unavailable");
     }
 
+    private static MethodHandle getIsVirtualThreadMethodHandle() {
+        if (JAVA_VERSION < 19) {
+            return null;
+        }
+        try {
+            MethodHandle methodHandle = MethodHandles.publicLookup().findVirtual(Thread.class, "isVirtual",
+                    MethodType.methodType(boolean.class));
+            // Call once to make sure the invocation works.
+            boolean isVirtual = (boolean) methodHandle.invokeExact(Thread.currentThread());
+            return methodHandle;
+        } catch (Throwable e) {
+            if (logger.isTraceEnabled()) {
+                logger.debug("Thread.isVirtual() is not available: ", e);
+            } else {
+                logger.debug("Thread.isVirtual() is not available: ", e.getMessage());
+            }
+            return null;
+        }
+    }
+
+    /**
+     * @param thread The thread to be checked.
+     * @return
+     * {@code true}: if this {@code thread} is a virtual thread.
+     */
+    static boolean isVirtualThread(Thread thread) {
+        // Quick exclusion.
+        if (thread == null || JAVA_VERSION < 19 || thread instanceof FastThreadLocalThread) {
+            return false;
+        }
+        // Uncommon case: `IS_VIRTUAL_THREAD_METHOD_HANDLE` is null.
+        if (IS_VIRTUAL_THREAD_METHOD_HANDLE == null) {
+            Class<?> clazz = thread.getClass();
+            if (clazz == Thread.class || (clazz = clazz.getSuperclass()) == Thread.class) {
+                return false;
+            }
+            return BASE_VIRTUAL_THREAD_CLASS_BINARY_NAME.equals(clazz.getName());
+        }
+        // Preferred fast check path:
+        try {
+            return (boolean) IS_VIRTUAL_THREAD_METHOD_HANDLE.invokeExact(thread);
+        } catch (Throwable t) {
+            // Should not happen.
+            if (t instanceof Error) {
+                throw (Error) t;
+            }
+            throw new Error(t);
+        }
```

Comment:
We can simplify this because the method handle being `null` implies the version check, and after optimization there's no overhead to the `invokeExact`, so there's no need to do the `instanceof` ourselves, and we can entirely avoid talking about the `BaseVirtualThread` class.

```suggestion
    /**
     * @param thread The thread to be checked.
     * @return {@code true}: if this {@code thread} is a virtual thread.
     */
    static boolean isVirtualThread(Thread thread) {
        // Quick exclusion.
        if (thread == null || IS_VIRTUAL_THREAD_METHOD_HANDLE == null) {
            return false;
        }
        // Check path:
        try {
            return (boolean) IS_VIRTUAL_THREAD_METHOD_HANDLE.invokeExact(thread);
        } catch (Throwable t) {
            // Should not happen.
            if (t instanceof Error) {
                throw (Error) t;
            }
            throw new Error(t);
        }
```

Reply:
Good point. Done the change.